src/Psy/TabCompletion/AutoCompleter.php

```diff
@@ -45,12 +45,13 @@ public function activate()
      * Handle readline completion.
      *
      * @param string $input Readline current word
-     * @param int    $index Current word index
+     * @param int    $start Readline start match
+     * @param int    $end   Readline end match
      * @param array  $info  readline_info() data
      *
      * @return array
      */
-    public function processCallback($input, $index, $info = array())
+    public function processCallback($input, $start, $end, $info = array())
```

Comment:
unfortunately this is a public method, so we won't be able to break backwards compatibility. right now.

but! `$info['end']` is supposed to be a thing already, and we're using it below. we could work the `$end` param from the callback into the `$info` array to augment and/or replace that one, or we could just use that one?

or we could add the `$end` param to the end here, but that's just a bit awkward.

or we could rename this method and add a deprecated stub of a method called `processCallback` that still takes the old arguments. i think i like this third option best :)

Reply:
It's actually the function passed to `readline_completion_function` that needs the change, which is also public. Refactoring that in a non-breaking manner gives me a large headache, so I reverted it back to the old (and admittedly broken) way. I use `point` now, which is more in line with what should happen anyway.